envi/__init__.py

```diff
@@ -24,6 +24,22 @@
 ARCH_THUMB       = 5 << 16
 ARCH_MSP430      = 6 << 16
 ARCH_H8          = 7 << 16
+ARCH_A64         = 8 << 16
+ARCH_RISCV32     = 9 << 16
+ARCH_RISCV64     = 10 << 16
+ARCH_PPC_E32     = 11 << 16
+ARCH_PPC_E64     = 12 << 16
+ARCH_PPC_S32     = 13 << 16
+ARCH_PPC_S64     = 14 << 16
+ARCH_PPCVLE      = 15 << 16
+ARCH_PPC_D       = 16 << 16
+ARCH_MCS51       = 17 << 16
+ARCH_RXV2        = 18 << 16
+ARCH_SPARC       = 19 << 16
+ARCH_SPARC64     = 20 << 16
+ARCH_MIPS32      = 21 << 16
+ARCH_MIPS64      = 22 << 16
```

Comment:
why do these have to be numbers in the first place? global string constants or an Enum should have equivalent performance and less likelihood of collisions. i'd only expect number constants like this when they might be used as indices into tables or with arithmetic/bitwise operations.

Reply:
[enum.StrEnum](https://docs.python.org/3/library/enum.html#enum.StrEnum) would be the modern Pythonic way. the existing code is very C-like. imho.

Reply:
currently, they *are* indexes into a list. :)
we could move to enums, but it wouldn't save us much.
however, it would save us collisions in merging.  it's worth considering.  maybe for v2.0.  thanks @williballenthin